common/src/chain/config/mod.rs

```diff
@@ -587,30 +588,74 @@ fn create_testnet_genesis() -> Genesis {
     )
 }
 
-pub fn regtest_genesis_values() -> (
+#[derive(Clone, Debug)]
+pub struct GenesisStakingSettings(String);
+
+impl GenesisStakingSettings {
+    pub fn new(settings: String) -> Self {
+        Self(settings)
+    }
+
+    pub fn get_settings(&self) -> Vec<&str> {
+        self.0.split(',').collect()
+    }
+}
+
+impl Default for GenesisStakingSettings {
+    fn default() -> Self {
+        Self::new("".to_string())
+    }
+}
+
+impl FromStr for GenesisStakingSettings {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(Self::new(s.to_string()))
+    }
+}
```

Comment:
My other suggestion notwithstanding, when you have a method that cannot fail, the `std::convert::Infallible` (or equivalent) type should be used as the error type. If errors are possible, try to find or introduce an `enum` that reflects the possible error space.

Reply:
As above, I updated to Infallible, but will change to an enum.